Support for building on amd64 and fix an incompatibility with newer gcc and binutils.

--- invaders.orig/compile.sh
+++ invaders/compile.sh
@@ -3,8 +3,8 @@
 set -e
 set -v
 
-CFLAGS="-fno-builtin -nostdinc -O2 -I. -Wall -Werror"
-LDFLAGS="-nostdlib -Wl,-N -Wl,-Ttext -Wl,100000"
+CFLAGS="-fno-builtin -nostdinc -O2 -I. -Wall -Werror -m32"
+LDFLAGS="-nostdlib -Wl,-N -Wl,-Ttext -Wl,100000 -Wl,--build-id=none -m32"
 gcc $CFLAGS -c keyboard.c
 gcc $CFLAGS -c delay.c
 gcc $CFLAGS -c common.c
